sdk/couchbase-core/src/httpx/client.rs

```diff
@@ -0,0 +1,100 @@
+use async_trait::async_trait;
+use base64::Engine;
+use base64::prelude::BASE64_STANDARD;
+use crate::httpx::base::{Request, Response, ResponseProvider};
+use crate::httpx::error::ErrorKind::{Connect, Generic};
+use crate::httpx::error::Result as HttpxResult;
+use reqwest::{redirect::Policy, Client as ReqwestClient};
+use reqwest::header::{CONTENT_TYPE, USER_AGENT};
+
+#[derive(Debug, Clone)]
+pub struct Client<C> {
+    pub inner: C,
+}
+
+#[async_trait]
+pub trait ClientProvider {
+    type R: ResponseProvider;
+    fn new(/*tlsConfig*/) -> HttpxResult<Self> where Self: Sized;
```

Comment:
Could we file a rscbc for adding tls to http please?

Reply:
Created RSCBC-44 👍